Plutarch/DataRepr/Internal.hs

```diff
@@ -108,6 +107,12 @@ data PDataRecord (as :: [PLabeledType]) (s :: S) where
     PDataRecord ((name ':= x) ': xs) s
   PDNil :: PDataRecord '[] s
 
+instance {-# OVERLAPPABLE #-} PlutusType (PDataRecord l) where
+  type PInner (PDataRecord l) _ = PBuiltinList PData
+  pcon' (PDCons x xs) = pcons # pforgetData x # pto xs
+  pcon' PDNil         = pnil
+  pmatch' _ _ = undefined -- FIXME
```

Comment:
You could just do `error "you need a more specific PDataRecord"` lol

Reply:
that's exactly what I've been leaning towards after trying out a bunch of things that didn't work....

that might honestly be the best thing to do right now. This instance is only really useful for the `pto` anyway.

Reply:
in a dependently typed language we could pattern match on the `l` probably

Reply:
it's ok I heard GHC is landing `DependentHaskell` tomorrow